modules/backend/formwidgets/permissioneditor/assets/js/permissioneditor.js

```diff
@@ -21,6 +22,37 @@
     // EVENT HANDLERS
     // ============================
 
+    PermissionEditor.prototype.onPermissionTypeClick = function (ev) {
+        var $rows = $(ev.target).closest('tr').nextAll()
+        var index = $(ev.target).index()
+
+        var allChecked = true
+
+        for (let i = 0; i < $rows.length; i++) {
+            var $row = $rows.eq(i)
+            var $check = $row.find('td:nth-child(' + (index + 1) + ') input[type=radio], td:nth-child(' + (index + 1) + ') input[type=checkbox]').eq(0)
+
+            if ($check.length > 0) {
+                if (!$check[0].checked) {
+                    allChecked = false
+                    break
+                }
+            }
+        }
+
+        for (let i = 0; i < $rows.length; i++) {
+            var $row = $rows.eq(i)
+            var $check = $row.find('td:nth-child(' + (index + 1) + ') input[type=radio], td:nth-child(' + (index + 1) + ') input[type=checkbox]').eq(0)
+            if ($check.length > 0) {
+                if ($check.is('input[type=checkbox]')) {
+                    $check.prop('checked', !allChecked)
+                } else {
+                    $check.prop('checked', true)
+                }
+            }
+        }
```

Comment:
These look like they're duplicates of each other, can this logic not be simplified into one loop?

Reply:
It can't be simplified because we don't know if there was at least one option which is not checked before processing the whole options.

Another approach could be unchecking all options until there was an unchecked option and loop back to check previous unchecked options. But this approach is less easy to read.